sdk/search/Azure.Search.Documents/tests/Serialization/SearchFilterTests.cs

```diff
@@ -178,30 +174,29 @@ public void Polygons()
                 "geo.intersects(Foo, geography'POLYGON((0 0,0 1,1 1,0 0))')",
                 SearchFilter.Create($"geo.intersects(Foo, {line})"));
 
-            GeoPolygon polygon = new GeoPolygon(new[] { line });
-            Assert.AreEqual(
-                "geo.intersects(Foo, geography'POLYGON((0 0,0 1,1 1,0 0))')",
-                SearchFilter.Create($"geo.intersects(Foo, {polygon})"));
+            //GeoPolygon polygon = new GeoPolygon(new[] { line });
+            //Assert.AreEqual(
+            //    "geo.intersects(Foo, geography'POLYGON((0 0,0 1,1 1,0 0))')",
+            //    SearchFilter.Create($"geo.intersects(Foo, {polygon})"));
 
             Assert.Throws<ArgumentException>(() => SearchFilter.Create(
-                $"{new GeoLine(new[] { new GeoPosition(0, 0) })}"));
-            Assert.Throws<ArgumentException>(() => SearchFilter.Create(
-                $"{new GeoLine(new[] { new GeoPosition(0, 0), new GeoPosition(0, 0), new GeoPosition(0, 0), new GeoPosition(1, 1) })}"));
+                $"{new GeoLineString(new[] { new GeoPosition(0, 0) })}"));
             Assert.Throws<ArgumentException>(() => SearchFilter.Create(
-                $"{new GeoPolygon(new[] { line, line })}"));
+                $"{new GeoLineString(new[] { new GeoPosition(0, 0), new GeoPosition(0, 0), new GeoPosition(0, 0), new GeoPosition(1, 1) })}"));
+            //Assert.Throws<ArgumentException>(() => SearchFilter.Create(
+            //    $"{new GeoPolygon(new[] { line, line })}"));
         }
-#endif
 
         [TestCaseSource(nameof(GetMicrosoftSpatialPointsData))]
         public void MicrosoftSpatialPoints(string filter) =>
             Assert.AreEqual("geo.distance(geography'POINT(2 3)', Foo) < 3", filter);
 
         private static IEnumerable GetMicrosoftSpatialPointsData()
         {
-            GeographyPoint point = GeographyPoint.Create(3.0, 2.0);
+            GeoPoint point = new (2.0, 3.0);
```

Comment:
+1. It bugged me so much (and tripped me up several times) that `GeographyPoint` reversed params from serialized representation and, IIRC, the `GeographyPosition` ctor params.